Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.swift

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct AddCustomDynamicEntityDataSourceView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The point on the screen the user tapped.
+    @State private var tappedScreenPoint: CGPoint?
+    
+    /// The placement of the callout.
+    @State private var calloutPlacement: CalloutPlacement?
+    
+    /// A dynamic entity observation that was selected on the screen.
+    @State private var selectedObservation: DynamicEntityObservation?
+    
+    var body: some View {
+        MapViewReader { proxy in
+            MapView(map: model.map)
+                .onSingleTapGesture { screenPoint, _  in
+                    tappedScreenPoint = screenPoint
+                    
+                    // Hides the callout.
+                    calloutPlacement = nil
```

Comment:
Consider doing this when the identify operation doesn't return any results. The way it is now, the animation is clearly off:

![Simulator Screen Recording - iPhone 14 Pro - 2023-08-11 at 16 05 39](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/2257493/08f2c5ef-4985-4e39-91d0-cc467808bd0b)